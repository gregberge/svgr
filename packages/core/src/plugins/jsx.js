import unified from 'unified'
import parse from 'rehype-parse'
import vfile from 'vfile'
import hastToBabelAst from 'hast-util-to-babel-ast'
import { transformFromAstSync, createConfigItem } from '@babel/core'
import removeJSXAttribute from '../babelPlugins/removeJSXAttribute'
import removeJSXEmptyExpression from '../babelPlugins/removeJSXEmptyExpression'

export default (code, config, state = {}) => {
  const filePath = state.filePath || 'unknown'
  const hastTree = unified()
    .use(parse, {
      fragment: true,
      space: 'svg',
      emitParseErrors: true,
      duplicateAttribute: false,
    })
    .parse(vfile({ path: filePath, contents: code }))

  const babelTree = hastToBabelAst(hastTree)

  const { code: transformedCode } = transformFromAstSync(babelTree, code, {
    plugins: [
      createConfigItem([
        removeJSXAttribute,
        { attribute: 'xmlns' },
        'removeXmlns',
      ]),
      createConfigItem(removeJSXEmptyExpression),
    ],
    filename: filePath,
    babelrc: false,
    configFile: false,
  })

  return transformedCode
}
