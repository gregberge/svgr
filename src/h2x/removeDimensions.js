
const dimensionAttrs = ['width', 'height']

const filterDimensionAttrs = attr => !dimensionAttrs.includes(attr.name)

const removeDimensions = () => ({
  visitor: {
    JSXElement: {
      enter(path) {
        // Skip if not svg node
        if (path.node.name !== 'svg') return

        path.node.attributes = path.node.attributes.filter(filterDimensionAttrs)
      },
    },
  },
})

export default removeDimensions
