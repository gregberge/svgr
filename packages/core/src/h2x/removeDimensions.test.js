import jsx from 'h2x-plugin-jsx'
import { transform } from 'h2x-core'
import removeDimensions from './removeDimensions'

describe('removeDimensions', () => {
  it('should remove width and height', () => {
    const result = transform(
      `<svg width="20" height="20">
        <g stroke="#063855" stroke-width="2" />
      </svg>`,
      { plugins: [jsx, removeDimensions()] },
    )

    expect(result).toMatchSnapshot()
  })
})
