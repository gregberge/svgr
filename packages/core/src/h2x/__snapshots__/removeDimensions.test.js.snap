// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`removeDimensions should remove width and height 1`] = `
"<svg>
  <g stroke=\\"#063855\\" strokeWidth={2} />
</svg>
"
`;
